ToDoList/Services/TodoService.cs

```diff
@@ -57,4 +58,38 @@ public async Task<List<Todo>> Get()
 
         return todos;
     }
+
+    public async Task<Todo> Update(UpdateTodoDTO input)
+    {
+        var todo = await context.Todos.FirstOrDefaultAsync(x =>
+            x.Id == input.Id);
+
+        if (todo is null)
+        {
+            return null;
```

Comment:
Since it is a simple todo application, we don't need to throw not not-found exception. Only the status code is enough. If we are working on an enterprise application, we can leverage on Result Pattern. We should be careful with throwing of exception